### DISCUSSION TITLE
Variable is undeclared but can be used

### DISCUSSION THREAD

**Discussion by wangfpp** [webpack/webpack#15337] [Q&A] [answered]
# Bug report

https://codesandbox.io/s/adoring-lehmann-dvfqf?file=/src/main.js

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

### 变量未声明可以使用 
```js
const useEffectFn = (_) => {
  return {
    visibleVar: null
  };
};
console.log(visibleVar); // 这里应该直接报错  should throw error
let visibleVar = useEffectFn();
console.log(visibleVar);
```

**Comment by vankop** [comment, marked as answer]
Thank you for creating this issue. Could you provide your webpack config?

**Reply by wangfpp** [reply]
I am using vue-cli-service": "~4.5.0",

**Reply by vankop** [reply]
I can not help with vue.. please reduce your issue  to simple webpack config, thanks.
